core/src/apps/cardano/sign_tx.py

```diff
@@ -170,6 +236,50 @@ def _build_outputs(
     return result
 
 
+def _build_certificates(
+    keychain: seed.Keychain, certificates: List[CardanoTxCertificateType]
+) -> List[Tuple]:
+    result = []
+    for certificate in certificates:
```

Comment:
Is there any reason we have iteration here and not let caller handle it via list comprehension?

Reply:
I think I prefer the iteration hidden here. The same question applies to `build_inputs`, `build_outputs` and `build_withdrawals` (although `build_withdrawals` returns a dict). I tried changing them to list comprehension, but it doesn't seem much simpler to me. I think we could talk about this offline - I'm curious about what the benefits are.

Reply:
Discussed offline - we'll keep it as is, since it wouldn't necessarily result in simpler code and `build_withdrawals` would have to do dict comprehension, which is a bit less transparent than simple list comprehensions.